pvanalytics/system.py

```diff
@@ -1 +1,245 @@
+        clipped.
+    clip_max : float, default 10.0
+        If the percent of data flagged as clipped is greater than
+        `clip_max` then tracking cannot be determined and
+        :py:const:`Tracker.UNKNOWN` is returned.
+    envelope_quantile : float, default 0.995
+        Quantile used to determine the upper power or irradiance
+        envelope.
+    fit_median : boolean, default True
+        Perform a secondary fit with the median power or irradiance to
+        validate that the profile is consistent through the entire
+        data set.
+    median_r2_min : float, default 0.9
+        Minimum :math:`r^2` for a curve fit to the median power or
+        irradiance at each minute of the day (Applies only if
+        `fit_median` is True).
+    fit_params : dict or None, default None
+        Minimum r-squared for curve fits according to the fraction of
+        data with clipping. This should be a dictionary with tuple
+        keys and dictionary values. The key must be a 2-tuple of
+        ``(clipping_min, clipping_max)`` where the values specify the
+        minimum and maximum percent of data with clipping for which
+        the associated fit parameters are applicable. The values of
+        the dicationary are themselves dictionaries with keys
+        ``'fixed'`` and ``'tracking'``, which give the minimum
+        :math:`r^2` for the curve fits, and ``'fixed_max'`` which
+        gives the maximum :math:`r^2` for a quadratic fit if the
+        system appears to have a tracker.
+
+        If None :py:data:`PVFLEETS_FIT_PARAMS` is used.
+    envelope_min_fraction : float, default 0.05
+        After calucluating the power or irradiance envelope data less
+        than `envelope_min_fraction` times the maximum of the envelope
+        is removed. This excludes data from morning and evening that
+        may interfere with curve fitting.
+    median_min_fraction : float, default 0.025
+        After calucluating the median power or irradiance at each
+        minute data less than `median_min_fraction` times the maximum
+        is removed. This excludes data from morning and evening that
+        may interfere with curve fitting.
+
+    Returns
+    -------
+    Tracker
+        The tracking determined by curve fitting (FIXED, TRACKING, or
+        UNKNOWN).
+
+    Notes
+    -----
+    Derived from the PVFleets QA Analysis project.
+
+    See Also
+    --------
+
+    pvanalytics.features.orientation.tracking_nrel
+
+    pvanalytics.features.orientation.fixed_nrel
+
+    """
+    fit_params = fit_params or PVFLEETS_FIT_PARAMS
+    series_daytime = series[daytime]
+    clip_percent = (clipping[daytime].sum() / len(clipping[daytime])) * 100
+    march_september = series_daytime[
+        series_daytime.index.month.isin([5, 6, 7, 8])
+    ]
+    october_february = series_daytime[
+        series_daytime.index.month.isin([11, 12, 1, 2])
+    ]
+    if len(october_february) == 0 or len(march_september) == 0:
```

Comment:
This seasonal logic needs to be described in the docstring. It is clearly meant for the northern hemisphere. We might want to provide kwargs to control it: `winter_months`, `seasonal_split`, I am just tossing ideas out.

Reply:
I think `seasonal_split` with `([5, 6, 7, 8], [11, 12, 1, 2])` as a default could work. I would like to shy away from words like `winter` or `summer` to at least minimize the northern hemisphere bias.